dingo/core/result.py

```diff
@@ -595,27 +601,43 @@ def plot_corner(self, parameters=None, filename="corner.pdf"):
             (Default: None)
         filename : str
             Where to save samples.
+        truths : dict
+            Dictionary of truth values to include.
+
+        Other Parameters
+        ----------------
+        legend_font_size: int
+            Font size of the legend.
+
         """
         theta = self._cleaned_samples()
         # delta_log_prob_target is not interesting so never plot it.
         theta = theta.drop(columns="delta_log_prob_target", errors="ignore")
 
+        if "weights" in theta:
+            weights = theta["weights"]
+        else:
+            weights = None
         # User option to plot specific parameters.
         if parameters:
             theta = theta[parameters]
+            if truths is not None:
+                kwargs["truths"] = [truths.get(k) for k in theta.columns]
```

Comment:
Should this not be outside the previous `if`?
```suggestion
        if truths is not None:
            kwargs["truths"] = [truths.get(k) for k in theta.columns]
```